core/services/vrf/v2/integration_helpers_test.go

```diff
@@ -1736,3 +1737,145 @@ func testMaliciousConsumer(
 	}
 	require.Equal(t, 1, len(requests))
 }
+
+func testReplayOldRequestsOnStartUp(
```

Comment:
Just a general question. What is the difference between integration tests here and integration tests using CTF: https://github.com/smartcontractkit/chainlink/tree/develop/integration-tests
Why do we need both?

Reply:
integration tests here (`/` root package) uses simulated backend, which doesn't actually start a real blockchain and gives more control points about blockchain behavior (e.g. when to commit a block). Useful when you need to manipulate certain states about the blockchain. 

QA integration tests (`/integration-tests`) are meant to run against real chains. This is as close it gets to testing in production like environment. Worth noting that QA integration test also has a DEV mode, which starts a local blockchain that runs on its own, which is different from the simulated blockchain above.